lms/djangoapps/courseware/tests/test_date_summary.py

```diff
@@ -280,19 +283,31 @@ def test_course_end_date_after_course(self):
 
     # Tests Verified Upgrade Deadline Date Block
 
-    def check_upgrade_banner(self, banner_expected=True, include_url_parameter=True):
+    def check_upgrade_banner(
+            self,
+            banner_expected=True,
+            include_url_parameter=True,
+            check_course_id=None
+            ):
         """
         Helper method to check for the presence of the Upgrade Banner
         """
         url = reverse('info', args=[self.course.id.to_deprecated_string()])
         if include_url_parameter:
             url += '?upgrade=true'
         resp = self.client.get(url)
+        upgrade_cookie_name = 'show_upgrade_notification'
         expected_banner_text = "Give yourself an additional incentive to complete"
         if banner_expected:
             self.assertIn(expected_banner_text, resp.content)
+            if upgrade_cookie_name in self.client.cookies:
```

Comment:
If the banner is supposed to be shown, the cookie should always be present. So go ahead and remove the conditional, since it could mask a bug.